### PR TITLE
Feature: Select Tool Implementation Step1 - Mark selected pixels 

### PR DESCRIPTION
**Pull request by hunkim98**
🚀 [Related Issue: #28 ]

### Preview

https://user-images.githubusercontent.com/57612141/236849972-dc7642f8-14f2-4888-bd65-de05b1f0c724.mov

<br/>

## Mark selected pixels when brush tool is select

- **[🎨Component] Modify GridLayer and InteractionLayer to mark selected pixels**

  - Added `selectingArea` and `selectedArea` for marking selected pixels
    - When user is **selecting** the area with mousedown with SELECT brush tool, the `selectingArea` is set in Interaction Layer.
    - After a mouseup, the `selectingArea` transforms into a `selectedArea`, which is the area in the pixel grid that was included in the `selectingArea`. After it is transformed, the `selectingArea` becomes`null`.
    - If there is a `selectingArea` or a `selectedArea` the lines are rendered in the grid layer through the method `renderSelection`

- **[🔗Other] Remove render function calls in the methods of Grid Layers**

  - Rendering in the any canvas, including Grid Layer, should only happen when the `render` method is directly called from the `Editor` class. If the `render` method is called in other layers, it is difficult to track, and difficult to draw additional drawings on that specific canvas context when needed.

<br/>

## Notes

- This is the first step of implementing a Select Tool. I will later implement the logic for moving the selected pixels around.

<br/>

## Next Up?

- I will continue to develop this feature.

### CODE REVIEW

File: src/utils/position.ts [hunk 1]
```diff
@@ -183,6 +183,149 @@ export const getPixelIndexFromMouseCartCoord = (
...
+export const getSelectedPixelRegion = (
+  selectingArea: { startWorldPos: Coord; endWorldPos: Coord },
+  rowCount: number,
+  columnCount: number,
+  gridSquareLength: number,
+) => {
+  const leftTopPoint: Coord = {
+    x: -((columnCount / 2) * gridSquareLength),
+    y: -((rowCount / 2) * gridSquareLength),
+  };
```
Comment (Lee-Si-Yoon) [inline src/utils/position.ts:195]: A bit confused could you elaborate how this value becomes leftTopPoint?

**Review by Lee-Si-Yoon — COMMENTED** [review]
And also would be better if you sync with doc for easier demo

**Review by lerrybe — APPROVED** [review]
I propose the Pn rule to inform the importance of opinion proposals. I'll use it sometimes!
[Pn rule: banksalad](https://blog.banksalad.com/tech/banksalad-code-review-culture/#%EC%BB%A4%EB%AE%A4%EB%8B%88%EC%BC%80%EC%9D%B4%EC%85%98-%EB%B9%84%EC%9A%A9%EC%9D%84-%EC%A4%84%EC%9D%B4%EA%B8%B0-%EC%9C%84%ED%95%9C-pn-%EB%A3%B0)

**Comment by lerrybe** [comment]
It seems to be an essential and interesting brush tool. I'm excited and looking forward to this feature!

File: src/utils/position.ts [hunk 2]
```diff
@@ -183,6 +183,149 @@ export const getPixelIndexFromMouseCartCoord = (
...
+    selectedAreaTopLeft.y = selectingArea.endWorldPos.y;
+    selectedAreaBottomRight.x = selectingArea.startWorldPos.x;
+    selectedAreaBottomRight.y = selectingArea.startWorldPos.y;
+  }
+  const startDistanceFromLeft = selectedAreaTopLeft.x - leftTopPoint.x;
+  const startDistanceFromTop = selectedAreaTopLeft.y - leftTopPoint.y;
+  if (startDistanceFromLeft < 0) {
+    selectedRegionTopLeft.x = leftTopPoint.x;
+  } else {
+    if (startDistanceFromLeft > columnCount * gridSquareLength) {
```
[the remark this answers is not in the document]
Comment (hunkim98) [reply src/utils/position.ts:250]: I will change the method variable names to clarify their roles

**Comment by hunkim98** [comment]
Thank you all for your comments! I have solved some issues that you have raised! Unsolved issues can be discussed later on another PR!

**Pull request merged**